### DISCUSSION TITLE
Error importing WP site

### DISCUSSION THREAD

**Discussion by mattknowles** [GetPublii/Publii#1776] [Help]
Just getting started with Publii this morning. I figured I'd try to import a simple WP site to start with.

I got the following error and have no idea what this means or how to correct it.

```
Rendering process failed
An error occurred during rendering process:

Cannot read properties of undefined (reading '1')

TypeError: Cannot read properties of undefined (reading '1')
    at TemplateHelper.saveOutputPageFile (/Applications/Publii.app/Contents/Resources/app.asar/back-end/modules/render-html/helpers/template.js:144:71)
    at Renderer.generatePages (/Applications/Publii.app/Contents/Resources/app.asar/back-end/modules/render-html/renderer.js:1209:33)
    at Renderer.generateWWW (/Applications/Publii.app/Contents/Resources/app.asar/back-end/modules/render-html/renderer.js:242:18)
    at Renderer.renderFullPreview (/Applications/Publii.app/Contents/Resources/app.asar/back-end/modules/render-html/renderer.js:193:20)
    at Renderer.renderSite (/Applications/Publii.app/Contents/Resources/app.asar/back-end/modules/render-html/renderer.js:174:28)
    at Renderer.render (/Applications/Publii.app/Contents/Resources/app.asar/back-end/modules/render-html/renderer.js:119:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async process.<anonymous> (/Applications/Publii.app/Contents/Resources/app.asar/back-end/workers/renderer/preview.js:17:22)


```

**Comment by dziudek** [comment]
@mattknowles - do you have page with ID = 1 on your pages list?

**Reply by mattknowles** [reply]
No I don't.

**Reply by dziudek** [reply]
@mattknowles - that's interesting - could you share with me a WXR file used to create your website? I will address some issues in WXR importer over the weekend for Publii v.0.46.2 planned for the next week so maybe I will be able to solve this issue too :) You can share your WXR file to dziudek[at]gmail[dot]com if you doesn't want to share it publicly

**Comment by mattknowles** [comment]
The client’s site is a nsfw photography site if that is ok. Matt KnowlesAesthetic Design & Photographywww.aestheticdesign.com(707) 407-5451On Oct 26, 2024, at 12:45, Tomasz Dziuda ***@***.***> wrote:﻿
@mattknowles - that's interesting - could you share with me a WXR file used to create your website? I will address some issues in WXR importer over the weekend for Publii v.0.46.2 planned for the next week so maybe I will be able to solve this issue too :) You can share your WXR file to dziudek[at]gmail[dot]com if you doesn't want to share it publicly

**Reply by dziudek** [reply]
@mattknowles - I have accessed the mentioned website (I had to use VPN, because my country is blocked by cloudflare) and I do not see any nsfw content or I didn't check it enough ;) 

Anyway - one additional thing - under input/config/pages.config.json file do you have something? Could you paste content of this file?

**Comment by mattknowles** [comment]
Sorry, the site linked in that message is my site, that’s just part of my email signature on my phone.

The client’s site that I was trying to import is [...]

I’ve disabled Cloudflare on that one. Unfortunately due to the number of AI scrapers, other bots, and hackers, I’ve been locking down sites in an attempt to to prevent run away bandwidth usage. That’s one reason I decided to look at your program. I’ve been doing web design since 1995 and I loved hand coding tight and efficient web sites. WordPress is not only such a hacking target, but it has gotten incredibly inefficient as well. Just one example, a 400K script is loaded to make sure a password is strong. What a waste.


Matt Knowles
Aesthetic Design & Photography
(707) 407-5451
***@***.***

**Comment by alex-hempel** [comment]
@mattknowles 

I've had the same thing happen to me. Basically it looks like Publii gets a hiccup interpreting imported Wordpress pages as posts.

As a workaround, you can check the database in your Publii site folder, using a tool like [DB Browser for SQLite](https://sqlitebrowser.org/). Open the db.sqlite file in your sites/<your-site>/input folder with DB Browser, and then browse the posts table. Find the ID 1 in the table, and see what it belongs to. If it's nothing important (my import created some phantom pages I didn't even use in Wordpress), you can just delete that row from the table.

Try rendering again, and if it keeps throwing errors for different IDs, rinse and repeat until it doesn't.
